packages/@expo/cli/src/start/server/metro/createExpoStickyResolver.ts

```diff
@@ -0,0 +1,172 @@
+  const resolvedModules = await autolinking.findModulesAsync(opts);
+  return Object.fromEntries(
+    Object.entries(resolvedModules).map(([key, entry]) => [key, entry.path])
+  );
+};
+
+const getReactNativeConfigResolutions = async (
+  opts: AutolinkingSearchOptions
+): Promise<Record<string, string>> => {
+  const reactNativeConfigModule = getReactNativeConfigModule();
+  const configResult = await reactNativeConfigModule.createReactNativeConfigAsync({
+    ...opts,
+    // NOTE(@kitten): web will use ios here. This is desired since this function only accepts android|ios.
+    // However, we'd still like to sticky resolve dependencies for web
+    platform: opts.platform === 'android' ? 'android' : 'ios',
+  });
+  return Object.fromEntries(
+    Object.entries(configResult.dependencies).map(([key, entry]) => [key, entry.root])
+  );
+};
+
+interface PlatformModuleDescription {
+  platform: AutolinkingPlatform;
+  moduleTestRe: RegExp;
+  resolvedModulePaths: Record<string, string>;
+}
+
+const getPlatformModuleDescription = async (
+  projectRoot: string,
+  platform: AutolinkingPlatform
+): Promise<PlatformModuleDescription> => {
+  const searchOptions = await getAutolinkingOptions(projectRoot, platform);
+  const autolinkingResolutions$ = getAutolinkingResolutions(searchOptions);
+  const reactNativeConfigResolutions$ = getReactNativeConfigResolutions(searchOptions);
+  const resolvedModulePaths = {
+    ...(await reactNativeConfigResolutions$),
+    ...(await autolinkingResolutions$),
+  };
+  const resolvedModuleNames = Object.keys(resolvedModulePaths);
+  debug(
+    `Sticky resolution for ${platform} registered ${resolvedModuleNames.length} resolutions:`,
+    resolvedModuleNames
+  );
+  return {
+    platform,
+    moduleTestRe: dependenciesToRegex(resolvedModuleNames),
+    resolvedModulePaths,
+  };
+};
+
+export type StickyModuleResolverInput = {
+  [platform in AutolinkingPlatform]?: PlatformModuleDescription;
+};
+
+export async function createStickyModuleResolverInput({
+  platforms,
+  projectRoot,
+}: {
+  projectRoot: string;
+  platforms: string[];
+}): Promise<StickyModuleResolverInput> {
+  return Object.fromEntries(
+    await Promise.all(
+      platforms
+        .filter((platform): platform is AutolinkingPlatform =>
+          AUTOLINKING_PLATFORMS.includes(platform as any)
+        )
+        .map(async (platform) => {
+          const platformModuleDescription = await getPlatformModuleDescription(
+            projectRoot,
+            platform
+          );
+          return [platformModuleDescription.platform, platformModuleDescription] as const;
+        })
+    )
+  ) as StickyModuleResolverInput;
+}
+
+export function createStickyModuleResolver(
+  input: StickyModuleResolverInput | undefined,
+  { getStrictResolver }: { getStrictResolver: StrictResolverFactory }
+): ExpoCustomMetroResolver | undefined {
+  if (!input) {
+    return undefined;
+  }
+
+  const fileSpecifierRe = /^[\\/]|^\.\.?(?:$|[\\/])/i;
```

Comment:
not sure if the regexp supports windows paths?

Reply:
(This is the same as the `createExpoFallbackResolver`)

These are specifier inputs i.e. import/require requests, which aren't in either POSIX/Win32 format specifically, but are instead specifiers for file paths, i.e. they should always be in POSIX format, since they're not file paths, but import paths